src/material/datepicker/calendar.ts

```diff
@@ -299,6 +305,9 @@ export class MatCalendar<D> implements AfterContentInit, AfterViewChecked, OnDes
   /** Whether the calendar is in month view. */
   get currentView(): MatCalendarView { return this._currentView; }
   set currentView(value: MatCalendarView) {
+    if (this._currentView !== value) {
+      this.viewChanged.emit(value);
```

Comment:
If we emit here, the event will come before the UI has actually updated. I think people would expect this event to come after the view has updated. I _think_ a `Promise.resolve().then()` should make this emit after, if not maybe a `setTimeout()`. Please also add a test that verifies this

Reply:
Event emitters have a `isAsync` flag which we could use here? E.g. `new EventEmitter(true)`.

Reply:
@fhuschle can you try removing the `setTimeout` and passing `true` when constructing the `EventEmitter`? If that works we should use it.

Reply:
That seems to work the same way.